File: 0199_binaryTreeRightSideView/main.cpp
#include <vector>
#include <queue>

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    std::vector<int> rightSideView(TreeNode* root) {
        std::vector<int> res;
        if(!root)
            return res;

        std::queue<TreeNode*> q;
        q.push(root);

        while(!q.empty()) {
            int size = q.size();
            res.push_back(q.back()->val);
            for(int i = 0; i < size; ++i) {
                TreeNode* curPtr = q.front();
                q.pop();
                if(curPtr->left)
                    q.push(curPtr->left);
                if(curPtr->right)
                    q.push(curPtr->right);
            }
        }
        return res;
    }
};

File: 0199_binaryTreeRightSideView/main.py
from typing import Optional, List
from collections import deque

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        queue: deque[Optional[TreeNode]] = deque()
        queue.append(root)
        res = []
        while queue:
            size = len(queue)
            for i in range(size):
                curNode = queue.popleft()
                if curNode.left:
                    queue.append(curNode.left)
                if curNode.right:
                    queue.append(curNode.right)
                if i == size - 1:
                    res.append(curNode.val)
        return res


        